replications/internal/queue_management.go

```diff
@@ -114,72 +119,73 @@ func (rq *replicationQueue) Close() error {
 	return rq.queue.Close()
 }
 
+// WriteFunc is currently a placeholder for the "default" behavior
+// of the queue scanner sending data from the durable queue to a remote host.
+func WriteFunc(b []byte) error {
+	return nil
+}
+
 func (rq *replicationQueue) run() {
 	defer rq.wg.Done()
 
-	retryInterval := time.Second
-	retryTimer := time.NewTicker(retryInterval)
-	defer retryTimer.Stop()
-
-	writer := func() {
-		for {
-			_, err := rq.SendWrite(func(b []byte) error {
-				rq.logger.Info("written bytes", zap.Int("len", len(b)))
-				return nil
-			})
-			if err != nil {
-				if err == io.EOF {
-					// No more data
-					// Handle this gracefully, as it is an expected error to receive
-				} else {
-					// todo more error handling
-					panic(1)
-				}
-				break
-			}
-		}
-	}
-
 	for {
 		select {
 		case <-rq.done: // end the goroutine when done is messaged
 			return
-		case <-retryTimer.C: // run the scanner every 1s
-			writer()
 		case <-rq.receive: // run the scanner on data append
-			writer()
+			for rq.SendWrite(rq.writeFunc) {
+			}
 		}
 	}
 }
 
-func (rq *replicationQueue) SendWrite(dp func([]byte) error) (int, error) {
-	// err here can be io.EOF, indicating nothing to write
+// SendWrite processes data enqueued into the durablequeue.Queue.
+// SendWrite is responsible for processing all data in the queue at the time of calling.
+// Retryable errors should be handled and retried in the dp function.
+// Unprocessable data should be dropped in the dp function.
+// An error from this function (other than io.EOF) will crash the InfluxDB server instance.
```

Comment:
This line of the comment (146) is no longer relevant I don't think, since we're returning a bool here / not crashing the server.